tutorial-review.md

```diff
@@ -0,0 +1,1186 @@
+{
+  "name": "mongo-blog",
+  "version": "0.0.0",
+  "private": true,
+  "scripts": {
+    "dev": "nodemon ./bin/www",
+    "start": "node ./bin/www",
++   "build-client": "cd ./client && yarn build"
+  },
+  "dependencies": {
+    "cookie-parser": "~1.4.4",
+    "cors": "^2.8.5",
+    "debug": "~2.6.9",
+    "express": "~4.16.1",
+    "http-errors": "~1.6.3",
+    "jade": "~1.11.0",
+    "mongoose": "^6.2.3",
+    "morgan": "~1.9.1"
+  },
+  "devDependencies": {
+    "nodemon": "^2.0.15"
+  }
+}
+```
+
+Let's run `npm install` in the terminal and we can now move to the client setup.
+
+First, at the root of your project directory create a folder `/client`, move into this folder and install React using `create-react-app`:
+
+```bash
+npx create-react-app .
+```
+
+Similarly to `express-generator` the command will create a ready-to-go React project hiding most of the tedious configurations required in the past.
+
+On top of the basic packages (react, react-dom...) we have to carefully think about what our blog client needs:
+
+- Clearly, the client should make API calls to the server to perform basic CRUD operations on the database.
+- There are gonna be different pages to create, read, edit and delete blog posts.
+- We can assume there is need for forms to create and edit a post.
+
+Those are very common functionalities and the npm ecosystem offers tons of different packages. For the purpose of the tutorial, we are gonna install `axios` to make API calls, `react-router-dom` to handle client routing and `react-hook-form` to submit form data.
+
+In the terminal, go ahead and install them:
+
+```bash
+yarn add axios react-router-dom react-hook-form
+```
+
+As both server and client share the same repo, the folder `/public` in the root will be used to return the static client after building it. To do so, we need to tweak the "build" script inside `/client/package.json` to build the static files in it:
+
+```javascript
+{
+  "name": "mongo-blog-client",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^12.1.3",
+    "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.26.0",
+    "bootstrap": "5.1.3",
+    "react": "^17.0.2",
+    "react-bootstrap": "^2.1.2",
+    "react-dom": "^17.0.2",
+    "react-hook-form": "^7.27.1",
+    "react-router-dom": "^6.2.1",
+    "react-scripts": "5.0.0",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
++   "build": "BUILD_PATH='../public' react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
+  "browserslist": {
+    "production": [
+      ">0.2%",
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  }
+}
+```
+
```

Comment:
What do you think about adding this line here:

"Finally, we are going to drop one file to prepare for our deployment: `package-lock.json`. From your project's root directory:

```
rm package-lock.json
```
"

Reply:
yeah sure